src/worker.js

```diff
@@ -0,0 +1,14 @@
+import { runAsWorker } from 'synckit';
+
+runAsWorker(async ({
+  text, prettierEslintPath, filePath, extensionConfig,
+}) => {
+  // eslint-disable-next-line import/no-dynamic-require, global-require
+  const format = require(prettierEslintPath);
```

Comment:
1. I don't think you need to use `require` here
2. My first thought is: we don't need any worker, just use `await import('prettier')` in the main thread if sync is not required

Reply:
@JounQin Dynamic imports are not supported by VS Code. I get the error `A dynamic import callback was not specified` when I try to use a dynamic import even with the hack. As you will see in this comment https://github.com/prettier/prettier-vscode/issues/3007#issuecomment-1596203079, it was fixed by using worker threads.

Reply:
OK, sorry I'm lacking of VSCode skills.

Reply:
But it's already inside a worker now, so `import()` should be supported here?

Reply:
It's got me confused too because I tried it and it caused the extension to hang.

Reply:
Do you mean with `synckit` or without? You can use the `SYNCKIT_TIMEOUT` environment or `timeout` option to debug.

Reply:
@JounQin I tried the timeout option and I get `Error: Internal error: Atomics.wait() failed: timed-out`
I'm trying this:
```javascript
const format = await import(prettierEslintPath);
  return format.default({
    text,
    filePath,
    prettierLast: extensionConfig?.prettierLast,
  });
```

Reply:
I'll help to debug today later.

Reply:
`prettierPath` could be ESM, so I think we should fix it before releasing.

Reply:
I tiny guess is that you're using esbuild to bundle this file as commonjs, this is not supported, so you should use `worker.mjs` instead.

Reply:
Not bundling it. I tried 'worker.mjs' and just copying the file directly to the 'dist' folder.

Reply:
<img width="215" alt="image" src="https://github.com/idahogurl/vs-code-prettier-eslint/assets/8336744/1b62c284-6394-4596-b35c-8ab32cb6a636">

I didn't see any `vscode-prettier-eslint` related output after changing codes and re-packaging + installing from `.vsix`.

Reply:
<img width="716" alt="image" src="https://github.com/idahogurl/vs-code-prettier-eslint/assets/8336744/bd2e7b1b-c363-4849-badd-f0b273a8849f">

I'm not a VSCode expert, any idea?

Reply:
See #201 @idahogurl